pkg/registry/apis/query/client/plugin.go

```diff
@@ -80,18 +80,8 @@ func (d *pluginClient) QueryData(ctx context.Context, req data.QueryDataRequest)
 		return http.StatusBadRequest, nil, err
 	}
 
-	code := http.StatusOK
 	rsp, err := d.pluginClient.QueryData(ctx, qdr)
-	if err == nil {
-		for _, v := range rsp.Responses {
-			if v.Error != nil {
-				code = http.StatusMultiStatus
-				break
-			}
-		}
-	} else {
-		code = http.StatusInternalServerError
-	}
+	code := query.GetResponseCode(rsp)
```

Comment:
Nit. If there's an `err` above the line is the `err` respected so that the query services returns a 500 response code in that case?

Reply:
that's a good point, i think i will just add it back again.

Reply:
added it back again

Reply:
👍